chemprop/cli/train.py

```diff
@@ -614,7 +661,8 @@ def main(args):
 
     if test_loader is not None:
         if args.task_type == "regression":
-            model.loc, model.scale = float(scaler.mean_), float(scaler.scale_)
+            model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(-1, 1))
+            model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(-1, 1))
```

Comment:
I think right now we leave transparent output transform for a future feature. That is, we should _not_ do transformation inside the `Predictor` and instead opt to untransform using a separate `StandardScaler`.

Reply:
@davidegraff Thanks for the comment! Wouldn't this requires users to have access to the training data, or maybe we need to add the `StandardScaler` in the checkpoint file somehow, so that users of the model can run inference and obtain what they expect?

Reply:
not to the checkpoint (as it's not needed during training), but in the saved model output directory. I'm generally a fan of making these separate files:
```
output_no_cv/
├── model_0/
[...]
└── model_*
    ├── model.ckpt
    ├── preds.csv
    ├── input_scaler.pkl
    └── output_scaler.pkl
```

Reply:
I see, that makes sense. I will remove them from this PR and open an issue to keep track of it.

Reply:
Actually, I have a question. Regarding this part of code that test the model, if the transformation is not done inside the predictor, then we should transform the test set. The reported results here would be in the transformed space but not the original space with physical meaning. Is that what we intend to do?

```
    if test_loader is not None:
        if args.task_type == "regression":
            model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(-1, 1))
            model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(-1, 1))
        results = trainer.test(model, test_loader)[0]
        logger.info(f"Test results: {results}")
```

Reply:
If I understand you correctly, the raw model output is in the standardized range, so you will need to use the `StandardScaler` to un-transform them back into the "physical" range. My general opinions which I think most people share:
- validation results should be comparable to training results, therefore we should **not** do any un-transformation
- testing results should be physically interpretable, therefore we _should_ un-transform.